Fatras/include/ActsFatras/Kernel/InteractionList.hpp

```diff
@@ -0,0 +1,344 @@
+/// multiple scattering or ionisation. Continous process types **must** provide
+/// a call operator with the following signature:
+///
+///     bool
+///     operator()(
+///         generator_t& rng,
+///         const Acts::MaterialSlab& slab,
+///         Particle& particle,
+///         std::vector<Particle>& generatedParticles) const
+///
+/// If multiple continuous processes are defined, they are executed serially in
+/// the order in which they are given.
+///
+/// For point-like processes, the passed material only affects the probability
+/// with which they occur but not the interaction itself, e.g. photon conversion
+/// into electron pairs. They are simulated by first drawing a limit on the
+/// material paths and then executing the interaction with the shortest limit
+/// when the drawn amount of material has been passed. Point-like process
+/// types **must** provide the following two member functions:
+///
+///     // generate X0/L0 limits
+///     std::pair<Scalar, Scalar>
+///     generatePathLimits(
+///         generator& rng,
+///         const Particle& particle) const
+///
+///     // run the process simulation
+///     bool
+///     run(
+///         generator_t& rng,
+///         Particle& particle,
+///         std::vector<Particle>& generatedParticles) const
+///
+/// For both cases, the output particle is modified in-place (if needed) and the
+/// return value indicates a break condition in the simulation, i.e. the
+/// particle is dead (true) or alive (false) after the interaction.
+///
+/// @note If an interaction destroys the incoming particle, the process
+///   simulation should indicate this via the break condition only and not
+///   by reducing the particle momentum to zero. The incoming particle should
+///   retain its initial kinematic state; the final kinematic state before
+///   destruction is typically of more interest to the user and this simplifies
+///   validation.
+///
+/// The physics processes are extendable by the user to accomodate their
+/// specific requirements. While the set of available physics processes must be
+/// configured at compile-time, within that set, processes can again be
+/// selectively disabled at run-time. By default all processes are applied.
+template <typename... processes_t>
+class InteractionList {
+  using Mask = std::bitset<sizeof...(processes_t)>;
+  using Processes = std::tuple<processes_t...>;
+  using ContinuousIndices = detail::ContinuousIndices<Processes>;
+  using PointLikeIndices = detail::PointLikeIndices<Processes>;
+
+ public:
+  /// Point-like interaction selection.
+  struct Selection {
+    Particle::Scalar x0Limit =
+        std::numeric_limits<Particle::Scalar>::infinity();
+    Particle::Scalar l0Limit =
+        std::numeric_limits<Particle::Scalar>::infinity();
+    size_t x0Process = SIZE_MAX;
+    size_t l0Process = SIZE_MAX;
+  };
+
+  /// Disable a specific process identified by index.
+  void disable(size_t process) { m_mask.set(process); }
```

Comment:
So the mask contains `1` for **disabled** processes? Is there a reason for this, because it's a bit counterintuitive I guess.